bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -247,6 +225,10 @@ public boolean readCharacteristic(BluetoothCharacteristic characteristic) {
             throw new IllegalStateException("TinyB device is not yet set");
         }
         BluetoothGattCharacteristic c = getTinybCharacteristicByUUID(characteristic.getUuid().toString());
+        if (c == null) {
+            logger.warn("Characteristic '{}' is missing on device '{}'.", characteristic.getUuid(), address);
```

Comment:
This was the line where we said on your older PR that the exception should be added, right?

Reply:
These null characteristic checks and logging were present in the enableNotifications and disableNotifications methods. They were missing in the `writeCharacteristic` and `readCharacteristic` methods so I added them.